src/app/pages/storage/modules/pool-manager/store/pool-manager.store.ts

```diff
@@ -117,6 +137,143 @@ export class PoolManagerStore extends ComponentStore<PoolManagerState> {
     }),
   );
 
+  disknumErrorMessage = helptext.manager_disknumErrorMessage;
+  disknumErrorConfirmMessage = helptext.manager_disknumErrorConfirmMessage;
+  disknumExtendConfirmMessage = helptext.manager_disknumExtendConfirmMessage;
+  vdevtypeErrorMessage = helptext.manager_vdevtypeErrorMessage;
+  exportedPoolsWarning = helptext.manager_exportedDisksWarning;
+
+  readonly poolCreationErrors$ = combineLatest([
+    this.wizardRequiredStepsValidity$,
+    this.topology$,
+  ])
+    .pipe(
+      map(([requiredSteps, topology]) => {
+        const errors: PoolCreationError[] = [];
+
+        Object.keys(requiredSteps).forEach((key: PoolCreationWizardStep) => {
+          const control = requiredSteps[key as PoolCreationWizardRequiredStep];
+          if (control.valid !== null && !control.valid && control.required) {
+            if (key === PoolCreationWizardStep.General) {
+              errors.push({
+                text: this.translate.instant('Name not added'),
+                severity: PoolCreationSeverity.Error,
+                step: key,
+              });
+            }
+            if (key === PoolCreationWizardStep.EnclosureOptions) {
+              errors.push({
+                text: this.translate.instant('No Enclosure selected for a Limit Pool To A Single Enclosure.'),
+                severity: PoolCreationSeverity.Error,
+                step: key,
+              });
+            }
+            if (key === PoolCreationWizardStep.Data) {
+              errors.push({
+                text: this.translate.instant('At least 1 data vdev is required.'),
+                severity: PoolCreationSeverity.Error,
+                step: key,
+              });
+            }
+          }
+        });
+
+        const nonEmptyTopologyCategories = this.filterNonEmptyCategories(topology);
+
+        nonEmptyTopologyCategories.forEach(([typologyCategoryType, typologyCategory], i) => {
+          let dataVdevDisknum = 0;
+          let dataVdevType: string;
+          let firstDataVdevType: CreateVdevLayout;
+          let firstDataVdevDisknum = 0;
+
+          if (typologyCategoryType === VdevType.Data) {
+            if (i === 0) {
+              firstDataVdevType = typologyCategory.layout;
```

Comment:
I think warnings related to editing a pool, such as `First vdev has {n} disks, new vdev has {m}` and `First vdev is a {vdevType}, new vdev is {newVdevType}` can be removed for now. We should prevent user from doing something like when editing.